AstronomyObjC/AstronomyObjC/Controllers/TMCFetchOperations.m

```diff
@@ -0,0 +1,79 @@
+//
+//  TMCFetchOperations.m
+//  AstronomyObjC
+//
+//  Created by Christopher Devito on 5/19/20.
+//  Copyright © 2020 Christopher Devito. All rights reserved.
+//
+
+#import "TMCFetchOperations.h"
+#import "TMCMarsPhotoReference.h"
+#import "NSURL+TMCUsingHTTPS.h"
+
+@interface TMCFetchOperations();
+
+@property (nonatomic, copy, nullable) NSData* imageData;
+@property TMCMarsPhotoReference *photo;
+@property (nonatomic, readwrite) NSURLSessionDataTask *task;
+typedef NS_ENUM(int, State) {isReady, isExecuting, isFinished};
+@property State state;
+
+@end
+
+@implementation TMCFetchOperations
+- (instancetype)initWithPhotoReference:(TMCMarsPhotoReference *)photo{
+    self = [super init];
+    if (self) {
+        _photo = photo;
+        self.state = isReady;
+    }
+    return self;
+}
+
+- (BOOL)isAsynchronous {
+    return YES;
+}
+
+- (void)start {
+    self.state = isExecuting;
+    NSURL *url = self.photo.imageURL.usingHTTPS;
+    NSURLSessionTask *task = [NSURLSession.sharedSession dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+
+        if (error) {
+            self.fetchError = error;
+            self.state = isFinished;
+            return;
+        }
+
+        if (!data) {
+            self.state = isFinished;
+            return;
+        }
+
+        self.imageData = data;
+        self.state = isFinished;
+        return;
+    }];
+    [task resume];
+}
```

Comment:
I alluded to this yesterday, but this method could be simplified a bit:

```swift
//...
NSURLSessionTask *task = [NSURLSession.sharedSession dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {

    // both of these might be nil, or might not; either way, the result is the same
    self.fetchError = error;
    self.imageData = data;

    self.state = isFinished;
    
    return;
}];
//...
```

Reply:
`return;` would also be unnecessary as the last line. 

However, it could still be useful to catch missing data and create a missingDataError or looking at the request != 200 to generate an HTTP error to send back.